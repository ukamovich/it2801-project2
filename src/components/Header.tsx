import React from "react";
import ToggleDarkMode from "../styles/DarkmodeButton";
import robot2k from '../Images/robot2k.png'

type props = {
    theme: string;
    setTheme: (arg0: string) => void;
}

class Header extends React.Component<props> {
    
    //function to switch between darkMode and LightMode
    changeTheme = () => {
        if (this.props.theme === "light") {
            this.props.setTheme("dark");
        } else {
            this.props.setTheme("light");
        }
        localStorage.setItem("theme", (this.props.theme === "light") ? "dark" : "light")
    };

    render() {
        return (
            
            <header className="header2k">
                <div className = "robocont">
                <img src={robot2k} id ="robo" alt="Logo" />
                </div>
                <div id="title">
                    <h1>The Commit Grabber</h1>
                </div>
                <div id="darkButtonCont">
                    <ToggleDarkMode onClick={this.changeTheme} id="darkButton">
                        {this.props.theme === "light" ? "DarkMode" : "LightMode"}
                    </ToggleDarkMode>
                </div>

                



            </header>

            
        );
    }
}

export default Header