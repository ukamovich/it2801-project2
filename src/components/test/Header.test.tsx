import Header from '../Header';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

let props = {
  theme: "light",
  setTheme: (arg0: string) => {
      props.theme = arg0
  }
}

//Snapshot test
describe("Header", () => {
  it("renders correctly", () => {
      const tree = renderer.create(
          <Header theme={props.theme} setTheme={props.setTheme}/>
          ).toJSON();
          expect(tree).toMatchSnapshot();
  });
});

/* Declared because it is being used as a props in Header*/
function dummyFunction(arg0: any) {
    console.log(`${arg0}`);
}

test('test that the Header-component renders successfully', () => {
    render(<Header theme = "" setTheme = {dummyFunction}/>);
  });

  test('test if <h1> (TEAM43) is contained when being rendered', () => {
    const { getByText } = render(<Header theme = "" setTheme = {dummyFunction}/>);
    const linkElement = getByText(/The Commit Grabber/i);
    expect(linkElement).toBeInTheDocument();
  });

  

