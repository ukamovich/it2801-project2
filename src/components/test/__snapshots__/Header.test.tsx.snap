// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  className="header2k"
>
  <div
    className="robocont"
  >
    <img
      alt="Logo"
      id="robo"
      src="robot2k.png"
    />
  </div>
  <h1>
    The Commit Grabber
  </h1>
  <button
    className="sc-bdfBQB gwImaZ"
    onClick={[Function]}
  >
    DarkMode
  </button>
</header>
`;
